src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -77,8 +79,130 @@ public bool Equals(TagHelperAttribute other)
             return
                 other != null &&
                 string.Equals(Name, other.Name, StringComparison.OrdinalIgnoreCase) &&
-                Minimized == other.Minimized &&
-                (Minimized || Equals(Value, other.Value));
+                Structure == other.Structure &&
+                (Structure == HtmlAttributeStructure.Minimized || Equals(Value, other.Value));
+        }
+
+        /// <inheritdoc />
+        public void WriteTo(TextWriter writer, HtmlEncoder encoder) => WriteTo(Name, Value, Structure, writer, encoder);
+
+        /// <inheritdoc />
+        public void CopyTo(IHtmlContentBuilder destination) => CopyTo(Name, Value, Structure, destination);
+
+        /// <inheritdoc />
+        public void MoveTo(IHtmlContentBuilder destination) => CopyTo(destination);
```

Comment:
Hmm, maybe we do need `MoveTo()` to handle a `Value` that implements `IHtmlContentContainer`. Previous `CopyAttributesTo()` code did not do that but I don't remember why. @rynowak ?